### PR TITLE
aws-c-common: modernize & support CMakeDeps

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

File: recipes/aws-c-common/all/conanfile.py [hunk 1]
```diff
@@ -75,14 +81,18 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "aws-c-common"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "aws-c-common")
+        self.cpp_info.set_property("cmake_target_name", "AWS::aws-c-common")
```
Comment (toge) [inline recipes/aws-c-common/all/conanfile.py:85]: A question for modifying other recipes.
Do we need to add `set_property` to the component definition?
Example: `self.cpp_info.components["aws-common-lib"]`.

Comment (SpaceIm) [reply recipes/aws-c-common/all/conanfile.py:85]: Eventually to avoid to create an unwanted target. @czoido what do you think?

Comment (czoido) [reply recipes/aws-c-common/all/conanfile.py:85]: Yes, I hope that I'm not missing something, but probably it's better to declare the `cmake_target_name` property for the component as well. Doing this the unwanted target is not created, the target that the consumer will get is the one from the component in the end (there's a check that will avoid the target from the root cpp_info to be created). I think this is the less bad option, and enough to maintain the support for  `cmake_find_package_*` for some time. I'll update the PR with the migration guide and recommend explicitly to set the property on the component as well. cc/ @SpaceIm @toge

Comment (toge) [reply recipes/aws-c-common/all/conanfile.py:85]: Thank you for your information!

**Review by toge — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by czoido — APPROVED** [review]

**Pull request merged**